astropy/coordinates/earth.py

```diff
@@ -174,6 +179,73 @@ def from_geodetic(cls, lon, lat, height=0., ellipsoid=None):
         self._ellipsoid = ellipsoid
         return self.to(height.unit)
 
+    @classmethod
+    def of_site(cls, site_name):
+        """
+        Return an object of this class for a known observatory/site by name.
+
+        Note that this is intended as a quick convenience function, not a
+        fully-featured registry of observatories.
+
+        Parameters
+        ----------
+        site_name : str
+            Name of the observatory.
+
+        Returns
+        -------
+        site : This class (a `~astropy.coordinates.EarthLocation` or subclass)
+            The location of the observatory.
+
+        See Also
+        --------
+        get_site_names : the list of sites that this function can access
+        """
+        # need to import inside function to avoid circular dependencies
+        from .sites import get_site
+
+        try:
+            el = get_site(site_name, online=True)  # this is always an EarthLocation
+        except URLError:
+            warn(AstropyUserWarning(_NO_ONLINE_SITES_WARNING_MSG))
+            el = get_site(site_name, online=False)  # this is always an EarthLocation
+
+        if cls is el.__class__:
```

Comment:
I went with a slightly different implementation that actually uses the name from the db file (which is sometimes different), but basically the same idea.